src/test/common/sessionParsing.test.ts

```diff
@@ -0,0 +1,298 @@
+
+			assert.strictEqual(result.toolName, 'Run Bash command');
+			assert.strictEqual(result.invocationMessage, '$ npm test\nTest output here');
+			if (result.toolSpecificData && 'language' in result.toolSpecificData) {
+				assert.strictEqual(result.toolSpecificData.language, 'bash');
+				assert.strictEqual(result.toolSpecificData.commandLine.original, 'npm test');
+			}
+		});
+
+		it('should parse think tool calls', function () {
+			const toolCall = {
+				function: {
+					name: 'think',
+					arguments: '{}'
+				},
+				id: 'call_789',
+				type: 'function',
+				index: 0
+			};
+
+			const result = parseToolCallDetails(toolCall, 'I need to analyze this code');
+
+			assert.strictEqual(result.toolName, 'Thought');
+			assert.strictEqual(result.invocationMessage, 'I need to analyze this code');
+		});
+
+		it('should parse report_progress tool calls', function () {
+			const toolCall = {
+				function: {
+					name: 'report_progress',
+					arguments: '{"prDescription": "Updated the test files", "commitMessage": "feat: add new tests"}'
+				},
+				id: 'call_101',
+				type: 'function',
+				index: 0
+			};
+
+			const result = parseToolCallDetails(toolCall, '');
+
+			assert.strictEqual(result.toolName, 'Progress Update');
+			assert.strictEqual(result.invocationMessage, 'Updated the test files');
+			assert.strictEqual(result.originMessage, 'Commit: feat: add new tests');
+		});
+
+		it('should handle unknown tool types', function () {
+			const toolCall = {
+				function: {
+					name: 'unknown_tool',
+					arguments: '{"param": "value"}'
+				},
+				id: 'call_999',
+				type: 'function',
+				index: 0
+			};
+
+			const result = parseToolCallDetails(toolCall, 'some content');
+
+			assert.strictEqual(result.toolName, 'unknown_tool');
+			assert.strictEqual(result.invocationMessage, 'some content');
+		});
+
+		it('should handle malformed tool arguments', function () {
+			const toolCall = {
+				function: {
+					name: 'str_replace_editor',
+					arguments: '{"invalid": json}'
+				},
+				id: 'call_error',
+				type: 'function',
+				index: 0
+			};
+
+			const result = parseToolCallDetails(toolCall, '');
+
+			// Should fall back gracefully with empty args - goes to the 'else' branch which returns 'Edit'
+			assert.strictEqual(result.toolName, 'Edit');
+		});
+
+		it('should handle repository root paths correctly', function () {
+			const toolCall = {
+				function: {
+					name: 'str_replace_editor',
+					arguments: '{"command": "view", "path": "/home/runner/work/repo/repo/"}'
+				},
+				id: 'call_root',
+				type: 'function',
+				index: 0
+			};
+
+			const result = parseToolCallDetails(toolCall, '');
+
+			assert.strictEqual(result.toolName, 'Read repository');
+			assert.strictEqual(result.invocationMessage, 'Read repository');
+		});
+	});
+
+	describe('parseDiff()', function () {
+		it('should parse diff content correctly', function () {
+			const diffContent = `diff --git a/src/file.ts b/src/file.ts
```

Comment:
Disabled the hygiene script for this since I couldn't figre out a way to disable it for the line.
@alexr00 is there a way to do that?

Reply:
The best thing to do would be to add these strings as test fixtures under ./test/common/fixtures. Maybe in a file ./test/common/fixtures/sessionParsing.ts and just export them all as strings from there.